waltz-server/src/main/java/com/wepay/waltz/server/internal/Locks.java

```diff
@@ -37,15 +37,16 @@ public Locks(int size, int numHashFuncs, long defaultHighWaterMark) {
     /**
      * Returns True if the locks doesn't overlap with the locks that are currently in use, otherwise returns False.
      * Also acquires the write locks while returning True.
-     * @param request List of write locks and read locks to use.
+     * @param request Locks to use.
      * @return True if the locks doesn't overlap with the locks that are currently in use, otherwise returns False.
      */
     public boolean begin(LockRequest request) {
         synchronized (locks) {
-            if (begin(request.writeLocks) && begin(request.readLocks)) {
-                // Mark entries for write locks
+            if (begin(request.writeLocks) && begin(request.readLocks) && begin(request.appendLocks)) {
+                // Mark entries for write locks and append locks
                 mark(request.writeLocks);
-                numActiveLocks++;
+                mark(request.appendLocks);
```

Comment:
Nit: somewhat subtle that readLocks doesn't need to be marked but appendLocks does. perhaps worth a comment.